selfservice/strategy/passkey/passkey_strategy.go

```diff
@@ -0,0 +1,111 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package passkey
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/kratos/continuity"
+	"github.com/ory/kratos/driver/config"
+	"github.com/ory/kratos/hash"
+	"github.com/ory/kratos/identity"
+	"github.com/ory/kratos/selfservice/errorx"
+	"github.com/ory/kratos/selfservice/flow/login"
+	"github.com/ory/kratos/selfservice/flow/registration"
+	"github.com/ory/kratos/selfservice/flow/settings"
+	"github.com/ory/kratos/session"
+	"github.com/ory/kratos/ui/node"
+	"github.com/ory/kratos/x"
+	"github.com/ory/x/decoderx"
+)
+
+type strategyDependencies interface {
+	x.LoggingProvider
+	x.WriterProvider
+	x.CSRFTokenGeneratorProvider
+	x.CSRFProvider
+
+	config.Provider
+
+	continuity.ManagementProvider
+
+	errorx.ManagementProvider
+	hash.HashProvider
+
+	registration.HandlerProvider
+	registration.HooksProvider
+	registration.ErrorHandlerProvider
+	registration.HookExecutorProvider
+	registration.FlowPersistenceProvider
+
+	login.HooksProvider
+	login.ErrorHandlerProvider
+	login.HookExecutorProvider
+	login.FlowPersistenceProvider
+	login.HandlerProvider
+
+	settings.FlowPersistenceProvider
+	settings.HookExecutorProvider
+	settings.HooksProvider
+	settings.ErrorHandlerProvider
+
+	identity.PrivilegedPoolProvider
+	identity.ValidationProvider
+	identity.ActiveCredentialsCounterStrategyProvider
+	identity.ManagementProvider
+
+	session.HandlerProvider
+	session.ManagementProvider
+}
+
+type Strategy struct {
+	d  strategyDependencies
+	hd *decoderx.HTTP
+}
+
+func NewStrategy(d any) *Strategy {
```

Comment:
Why not `strategyDependencies`?

Reply:
This is the standard factory signature, so that we can pass an array of replacement strategies (which we use in Cloud). Otherwise, each factory would have a different signature.

See: https://github.com/ory/kratos/blob/62c6118bf2e6b3cd984bdda94c084848678e2c6a/driver/registry.go#L211-L220

Reply:
ah I see.